Cython/Compiler/UtilityCode.py

```diff
@@ -265,6 +267,60 @@ def filter_inherited_directives(current_directives):
         return utility_code_directives
 
 
+class CythonSharedUtilityCode:
+    def __init__(self, module_name, context, requires):
+        self._module_name = module_name
+        self.context = context
+        self.requires = requires
+        self.pxd_scope = None
+
+    def find_module(self, context):
+        import Cython
+        from .Scanning import FileSourceDescriptor
+        from .Errors import CompileError
+        qualified_name = '.'.join([Options.use_shared_utility, self._module_name])
+        scope = context
+        for name, is_package in scope._split_qualified_name(qualified_name, relative_import=False):
+            scope = scope.find_submodule(name, as_package=is_package)
+
+        pxd_pathname = os.path.join(
+            os.path.split(Cython.__file__)[0],
+            'Utility',
+            self._module_name + '.pxd'
+        )
+        try:
+            rel_path = qualified_name.replace('.', os.sep) + os.path.splitext(pxd_pathname)[1]
+            source_desc = FileSourceDescriptor(pxd_pathname, rel_path)
+            source_desc.in_utility_code = True
+            err, result = context.process_pxd(source_desc, scope, qualified_name)
+            (pxd_codenodes, pxd_scope) = result
+            context.utility_pxd = (pxd_codenodes, pxd_scope)
+            scope.pxd_file_loaded = True
+            if err:
+                raise err
+        except CompileError:
+            pass
```

Comment:
This seems very broad. I'm sure there are several cases in which we shouldn't just ignore the error. In which cases is it ok to do so? (If we keep the `try-except`, they seem worth documenting here.)